workflow-templates/review-release.yml

```diff
@@ -0,0 +1,54 @@
+name: Review Release
+concurrency:
+  group: app-release
+  cancel-in-progress: true
+permissions:
+  contents: read
+  id-token: write
+  statuses: read
+on:
+  workflow_dispatch:
+    inputs:
+      task_token:
+        description: 'StepFunction task token'
+        required: true
+
+jobs:
+  review_release:
+    runs-on: ubuntu-latest
+    outputs:
+      manual_review_required: ${{ steps.review_release.outputs.manual_review_required }}
+    steps:
+      - id: review_release
+        uses: 'phantomcyber/dev-cicd-tools/github-actions/review-release@main'
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+  manual_review:
+    needs: [review_release]
+    if: needs.review_release.outputs.manual_review_required == 'true'
+    runs-on: ubuntu-latest
+    environment: prod
+    steps:
+      - run: exit 0
+  approve_release:
+    needs: [review_release, manual_review]
+    if: always() && (needs.manual_review.result == 'success' || (needs.review_release.result == 'success' && needs.manual_review.result == 'skipped'))
```

Comment:
interesting, so the `always()` needs to be specified so it runs even on prior job failures despite the rest of the conditional?

Reply:
yep, the `needs` section effectively just makes sure the job is run after its dependent jobs, then we check their in the rest of the conditional